core/src/autogluon/core/utils/utils.py

```diff
@@ -40,40 +40,66 @@ def get_gpu_count():
     return gpu_count
 
 
-def generate_kfold(X, y=None, n_splits=5, random_state=0, stratified=False, n_repeats=1):
-    # TODO: Add GroupKFold
-    if stratified and (y is not None):
-        if n_repeats > 1:
-            kf = RepeatedStratifiedKFold(n_splits=n_splits, n_repeats=n_repeats, random_state=random_state)
+class CVSplitter:
+    def __init__(self,
+                 splitter_cls=None,
+                 n_splits=5,
+                 n_repeats=1,
+                 random_state=0,
+                 stratified=False,
+                 groups=None):
+        self.n_splits = n_splits
+        self.n_repeats = n_repeats
+        self.random_state = random_state
+        self.stratified = stratified
+        self.groups = groups
+        if splitter_cls is None:
+            splitter_cls = self._get_splitter_cls()
+        self._splitter = self._get_splitter(splitter_cls)
+
+    def _get_splitter_cls(self):
+        if self.groups is not None:
+            num_groups = len(self.groups.unique())
+            if self.n_repeats != 1:
+                raise AssertionError(f'n_repeats must be 1 when split groups are specified. (n_repeats={self.n_repeats})')
+            self.n_splits = num_groups
+            splitter_cls = LeaveOneGroupOut
```

Comment:
log a warning that the feature is experimental

Reply:
This would log spam: This is repeated for every bag.